File: Factory/FactoryMethod/Java/src/Pizza/NewYorkStyle/NewYorkStyleCheesePizza.java
package eu.svetlin.design.patterns.factory.factorymethod.Pizza.NewYorkStyle;

import eu.svetlin.design.patterns.factory.factorymethod.Pizza.Pizza;

public class NewYorkStyleCheesePizza extends Pizza {
    public NewYorkStyleCheesePizza(){
        name = "New York Style Sauce and Cheese Pizza";
        dough = "Thin Crust Dough";
        sauce = "Marinara Sauce";

        toppings.add("Grated Reggiano Cheese");
    }
}

File: Factory/FactoryMethod/TypeScript/src/Pizza/NewYorkStyle/NewYorkStyleCheesePizza.ts
import Pizza from '../Pizza';

class NewYorkStyleCheesePizza extends Pizza {
  constructor() {
    super();
    this.name = 'New York Style Sauce and Cheese Pizza';
    this.dough = 'Thin Crust Dough';
    this.sauce = 'Marinara Sauce';

    this.toppings.push('Grated Reggiano Cheese');
  }
}

export default NewYorkStyleCheesePizza;
